pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -0,0 +1,117 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package machinesync
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	corev1resourcebuilder "github.com/openshift/cluster-api-actuator-pkg/testutils/resourcebuilder/core/v1"
+	machinev1resourcebuilder "github.com/openshift/cluster-api-actuator-pkg/testutils/resourcebuilder/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/openshift/cluster-capi-operator/pkg/test"
+)
+
+var _ = Describe("MachineSync Reconciler", func() {
+	var mgrCancel context.CancelFunc
+	var mgrDone chan struct{}
+	var mgr manager.Manager
+	var reconciler *MachineSyncReconciler
+
+	var namespace *corev1.Namespace
+	var namespaceName string
+
+	var machineBuilder machinev1resourcebuilder.MachineBuilder
+	var machine *machinev1beta1.Machine
+
+	startManager := func(mgr *manager.Manager) (context.CancelFunc, chan struct{}) {
+		mgrCtx, mgrCancel := context.WithCancel(context.Background())
+		mgrDone := make(chan struct{})
+
+		go func() {
+			defer GinkgoRecover()
+			defer close(mgrDone)
+
+			Expect((*mgr).Start(mgrCtx)).To(Succeed())
+		}()
+
+		return mgrCancel, mgrDone
+	}
+
+	stopManager := func() {
+		mgrCancel()
+		// Wait for the mgrDone to be closed, which will happen once the mgr has stopped
+		<-mgrDone
+	}
+
+	BeforeEach(func() {
+		By("Setting up a namespace for the test")
+		namespace = corev1resourcebuilder.Namespace().WithGenerateName("control-plane-machine-set-controller-").Build()
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+		namespaceName = namespace.GetName()
+
+		By("Setting up the machine builder")
+		machineBuilder = machinev1resourcebuilder.Machine().
+			WithNamespace(namespaceName).
+			WithGenerateName("foo").
+			WithProviderSpecBuilder(machinev1resourcebuilder.AWSProviderSpec())
+
+		By("Setting up a manager and controller")
+		var err error
+		mgr, err = ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: testScheme,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Manager should be able to be created")
+
+		reconciler = &MachineSyncReconciler{
+			Client: mgr.GetClient(),
+		}
+		Expect(reconciler.SetupWithManager(mgr)).To(Succeed(), "Reconciler should be able to setup with manager")
+	})
+
+	AfterEach(func() {
+		Expect(test.CleanupAndWait(ctx, k8sClient, machine)).To(Succeed())
+	})
+
+	JustBeforeEach(func() {
+		By("Starting the manager")
+		mgrCancel, mgrDone = startManager(&mgr)
+	})
+
+	JustAfterEach(func() {
+		By("Stopping the manager")
+		stopManager()
+	})
+
+	It("should reconcile without erroring", func() {
+		machine = machineBuilder.Build()
+
+		_, err := reconciler.Reconcile(ctx, reconcile.Request{
```

Comment:
Calling reconcile directly is odd, but, fine for now I guess, I would imagine this test will be replaced with more specific ones when we work out what we need?